database.py

```diff
@@ -230,7 +230,7 @@ def __load_data(self):
 
     def __load_extra(self):
         """Load all the file names of the images in the Extra folder."""
-        for file in os.listdir(self.directory + "/./Images/Extra"):
+        for file in os.listdir(self.directory + "/Images/Extra"):
             if file.endswith(".jpg"):
                 name = os.path.join("/Images/Extra", file)\
                         .split('/')[-1][0:-4].lower()
```

Comment:
This looks great!  It is good to see that that tests are starting to help.

Could you please replace lines 233 to 236 with something like:
```python
extra_dir = os.path.join(self.directory, "Images", "Extra")
for file in os.listdir(extra_dir):
    name, ext in os.path.splitext(file.lower())
    if ext == ".jpg":
```

The code that is currently on lines 235 and 236 makes my head spin.